guacml/models/linear_model.py

```diff
@@ -1,38 +1,40 @@
-from sklearn.linear_model import LogisticRegression
+import pandas as pd
+import numpy as np
+import pickle
 from hyperopt import hp
+from sklearn.linear_model import LogisticRegression
 from sklearn.linear_model import Ridge
 
-from guacml.enums import ProblemType
 from guacml.models.base_model import BaseModel
-from guacml.preprocessing.column_analyzer import ColType
-import pandas as pd
-import numpy as np
 
 
 class LinearModel(BaseModel):
 
+    def copy_model(self):
+        return pickle.loads(pickle.dumps(self.model))
```

Comment:
did you consider using [`copy.deep_copy`](https://docs.python.org/3/library/copy.html#copy.deepcopy) or is it intentional that you go through pickle?